selfapi.js

```diff
@@ -367,6 +374,24 @@ API.prototype = {
     });
   },
 
+  // Build index routes.
+  buildIndexRoutes: function (basePath) {
+      var fullPath = normalizePath(this.path, basePath) || '/';
+      this.get({
+          title: 'Index',
+          handler: (request, response) => {
+              var routes = {};
+              Object.keys(this.children).forEach(child => {
+                  if (child == "null") {
```

Comment:
Whoa, in which case can a `child` actually be `"null"` string (or compatible type)? This sounds bad, and I think we shouldn't add such a check and instead let things fail so that we can fix any broken children (this sounds weird).